includes/core/server/request/class-create-and-confirm-intention.php

```diff
@@ -0,0 +1,217 @@
+<?php
+/**
+ * Class file for WCPay\Core\Server\Request\Create_Intent.
+ *
+ * @package WooCommerce Payments
+ */
+
+namespace WCPay\Core\Server\Request;
+
+use Exception;
+use WC_Payments;
+use WCPay\Core\Exceptions\Invalid_Request_Parameter_Exception;
+use WCPay\Core\Server\Request;
+use WC_Payments_API_Client;
+
+/**
+ * Request class for creating intents.
+ */
+class Create_And_Confirm_Intention extends Request {
+	use Intention;
+	use Level3;
+
+	const IMMUTABLE_PARAMS = [
+		// Those are up to us, we have to decide.
+		'amount',
+		'currency',
+		'payment_method',
+	];
+
+	const REQUIRED_PARAMS = [
+		'amount',
+		'currency',
+		'payment_method',
+		'customer',
+		'metadata',
+	];
+
+	const DEFAULT_PARAMS = [
+		'confirm'        => true, // By the definition of the request.
+		'capture_method' => 'automatic',
+	];
+
+	/**
+	 * Returns the request's API.
+	 *
+	 * @return string
+	 */
+	public function get_api(): string {
+		return WC_Payments_API_Client::INTENTIONS_API;
+	}
+
+	/**
+	 * Returns the request's HTTP method.
+	 */
+	public function get_method(): string {
+		return 'POST';
+	}
+
+	/**
+	 * Amount setter.
+	 *
+	 * @param  int $amount Amount to charge.
+	 * @return static      Instance of the class for method chaining.
+	 */
+	public function set_amount( int $amount ) {
+		$this->validate_is_larger_then( $amount, 0 );
+		return $this->set_param( 'amount', $amount );
+	}
+
+	/**
+	 * Currency code setter.
+	 *
+	 * @param  string $currency_code Currency to charge in.
+	 * @return static                Instance of the class for method chaining.
+	 * @throws Invalid_Request_Parameter_Exception When the currency code is invalid.
+	 */
+	public function set_currency_code( string $currency_code ) {
+		$this->validate_currency_code( $currency_code );
+
+		return $this->set_param( 'currency', $currency_code );
+	}
+
+	/**
+	 * Payment method setter.
+	 *
+	 * @param  string $payment_method_id ID of payment method to process charge with.
+	 * @return static                    Instance of the class for method chaining.
+	 */
+	public function set_payment_method( $payment_method_id ) {
+		$this->validate_stripe_id( $payment_method_id, [ 'pm', 'src' ] );
+		return $this->set_param( 'payment_method', $payment_method_id );
+	}
+
+	/**
+	 * Customer setter.
+	 *
+	 * @param  string $customer_id ID of the customer making the payment.
+	 * @return static              Instance of the class for method chaining.
+	 */
+	public function set_customer( string $customer_id ) {
+		$this->validate_stripe_id( $customer_id, 'cus' );
+		return $this->set_param( 'customer', $customer_id );
+	}
+
+	/**
+	 * Capture method setter.
+	 *
+	 * @param  bool $manual_capture Whether to capture funds via manual action.
+	 * @return static               Instance of the class for method chaining.
+	 */
+	public function set_capture_method( bool $manual_capture = false ) {
```

Comment:
The default value of `false` feel very counter-intuitive

Reply:
Good point, it just used to be the default in the API client method, and this was thoughtlessly carried over here.